hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/querymatcher/ScanQueryMatcher.java

```diff
@@ -263,9 +263,13 @@ public void clearCurrentRow() {
    * Set the row when there is change in row
    */
   public void setToNewRow(Cell currentRow) {
-    this.currentRow = currentRow;
-    columns.reset();
-    reset();
+    if (
+      this.currentRow == null || this.rowComparator.compareRows(currentRow, this.currentRow) != 0
```

Comment:
Why do we need to do another compareRows here? This is called in StoreScanner:

```
if ((matcher.currentRow() == null) || !CellUtil.matchingRows(cell, matcher.currentRow())) {
      this.countPerRow = 0;
      // The setToNewRow will call reset internally
      matcher.setToNewRow(cell);
    }
```

So we already are doing a compareRows

Reply:
Thanks for pointing out this. I think we should remove the row comparation in StoreScanner#resetQueryMatcher.

Reply:
Can you use CellUtil.matchingRows here like we were using in StoreScanner? That has an optimization where it checks the last byte of the array/buffer early. Comparing rows requires walking each array/buffer start to finish. The matchingRows is better because we are always comparing adjacent rows, where the earlier bits are more likely to be equal.

Reply:
So now I'm wondering.... what's the functional change here? Seems like you just moved the comparison into the method, but its all the same. Why do we need this change?